### ISSUE TITLE
Callback logic handling (code and tokens exchange) after authentication

### ISSUE THREAD

**Issue by absay** [damienbod/angular-auth-oidc-client#393]
**tl;dr:** after the authorization server returns the code, how to handle the response with this library?

---
When the callback URL is called after the Authorization endpoint provides the code (i.e. after successful authentication), is it possible to use this library for handling the logic?

So far, I'm using a Node.js server running on separate port (e.g. :5010) while the Angular app runs on a different one (.e.g :4200). The callback url registered for my app is the one on the Node.js app, which makes use of Passport to handle the rest of the authentication process: code and token exchanges, then it goes to the redirect page.

This architecture makes little sense and I was wondering if I could put the necessity of a separate "re-authentication" with Node.js/Passport completely out of the equation, so that the whole process can be handled within the Angular app itself. The docs do seem to mention something related but I'm really unclear on how to go about it.

Thanks in advance for any useful pointers and thanks as well for this library which is the one that implements the PKCE steps for Angular apps. 👍

**Comment by damienbod** [comment]
closing this, please open if you still have an issue
